DataStructuresMemories/ViewControllers/DataStructViewController.swift

```diff
@@ -8,49 +8,103 @@
 
 import UIKit
 
-class DataStructViewController: UIViewController {
+class DetailViewController: UIViewController {
     
     @IBOutlet weak var textInsideLable: UILabel!
     
-    var tappedCell: DataStruct?
+    var tappedCell: DataStructProtocol?
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.navigationItem.title = tappedCell?.title
+        if let title = tappedCell?.title {
+            self.navigationItem.title = title
+        }
```

Comment:
Below `gradientView.opacityGradient()`
Method should contain action ("construct", "add" etc).
Nouns are for properties